### PR TITLE
[BugFix] Detect empty distillation sequences in multidimensional batches

### LINKED ISSUE #4081: [BUG] DistillationLoss misses empty sequences in multidimensional batches
## Describe the bug

`DistillationLoss` checks whether each scored sequence has at least one selected token, but the check collapses every remaining dimension of each outer-batch slice. For a multidimensional TensorDict batch, an empty sequence therefore passes whenever another sequence in the same slice has a selected token.

The loss then silently returns zero for that sequence because sequence-length normalization clamps its zero denominator. This is particularly relevant to grouped LLM batches such as `[prompts, generations]`.

## To reproduce

On current `main` (`d7659c781a55f7680a025f16015064b0a8169344`):

```python
import torch
from tensordict import TensorDict
from tensordict.nn import TensorDictModule
from torchrl.objectives.llm import DistillationLoss

mask = torch.ones(2, 2, 3, dtype=torch.bool)
mask[0, 1] = False  # one complete sequence has no selected tokens
student_log_probs = torch.full((2, 2, 3), -1.0)
data = TensorDict(
    {
        (history, full): student_log_probs,
        (masks, all_attention_mask): mask,
        (masks, all_assistant_mask): mask,
        (next, teacher_log_probs, full): torch.full((2, 2, 3), -2.0),
    },
    batch_size=(2, 2),
)
student = TensorDictModule(
    torch.nn.Identity(),
    in_keys=[(history, full)],
    out_keys=[(log_probs, full)],
)

print(DistillationLoss(actor_network=student, reduction=none)(data).loss_distill)
```

Current output:

```text
tensor([[0.3679, 0.0000],
        [0.3679, 0.3679]])
```

## Expected behavior

Raise the existing `ValueError` saying that some sequences have no tokens selected. That is already the intended behavior for a one-dimensional batch; it should apply independently to every sequence in any supported TensorDict batch shape.

## Reason and possible fix

`forward` currently uses `not mask.any()` while iterating over the outer batch dimension. It should reduce only the token dimension, then check whether any resulting per-sequence flag is false. The token dimension is already encoded by the existing reductions as `tensordict.ndim - 1` after the outer iteration.

## System info

- Source checkout at `d7659c781a55f7680a025f16015064b0a8169344`
- TensorDict 0.13.0
- PyTorch 2.11.0+cu130 (CPU reproduction)
- Python 3.13.3
- Windows 11

## Checklist

- [x] I checked issues, pull requests, commits, and the introducing PR #4038 for similar reports.
- [x] I read the DistillationLoss documentation.
- [x] I provided a minimal working example.

## AI assistance disclosure

I used OpenAI Codex to help inspect the implementation, construct and run the reproduction, search for duplicates, and draft this report. I reviewed the reproduced behavior and report before submission.

### PR DESCRIPTION
**Pull request by aswanth-07**
## Description

Fixes `DistillationLoss`'s empty-sequence validation for multidimensional TensorDict batches. The previous `mask.any()` check collapsed both the token and remaining batch dimensions after the outer iteration, so an empty sequence was accepted whenever a sibling sequence selected at least one token.

The validation now reduces only the token dimension and checks every resulting per-sequence flag. The same token-dimension value is reused by the existing KL and sequence-length reductions. A transformers-free regression covers a `[2, 2]` batch where exactly one sequence is empty.

Closes #4081

## Motivation and Context

An accepted empty sequence silently received a zero loss because normalization clamps the zero token count. Grouped LLM batches such as `[prompts, generations]` could therefore hide malformed masks instead of raising the documented validation error.

- [x] I have raised an issue to propose this change: #4081

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

## Validation

- `python -m pytest test/llm/test_llm_objectives.py -q -k "distillation and not integration"`: 16 passed
- `python -m pytest test/llm/test_llm_objectives.py -q -m "not slow and not integration"`: 69 passed, 1 skipped, 2 deselected
- Repository pre-commit hooks on both changed files: all passed (ufmt, flake8, pydocstyle, pyupgrade, codespell, autoflake, docstring checks, and file checks)
- `git diff --check`: passed

## Checklist

- [x] I have read the CONTRIBUTING guide and CLAUDE.md.
- [x] I have updated the tests accordingly.
- [ ] My change requires a documentation update.
- [ ] I have updated the documentation accordingly.

## AI assistance disclosure

I used OpenAI Codex to help inspect the implementation, construct and run the pristine reproduction, search issues/pull requests/commits and the introducing PR for duplicates, implement the focused fix and regression, run validation, and draft this pull request. I reviewed the diff and reproduced the behavior before submission.

### CODE REVIEW

**Comment by vmoens** [comment]
Reviewed rebased head `0fe08b2dd1` plus the current PR discussion (there are no human review threads yet). The fix is correctly scoped: each `as_list=True` mask retains the remaining batch dimensions plus tokens, so reducing `dim=tensordict.ndim - 1` handles only the token axis and the outer `.any()` detects an empty sequence anywhere in the residual batch. Reusing the same axis for KL summation and normalization keeps those paths aligned. The new multidimensional regression passes locally (`1 passed`), and `git diff --check` is clean. No blocking finding from this review.

**Review by vmoens — APPROVED** [review]
LGTM
